### ISSUE TITLE
#17126 broke the `platform-serverless-vercel/vercel-cli` ecosystem tests

### ISSUE THREAD

**Issue by jkomyno** [prisma/prisma#17264]
- [x] we should understand why that's the case
- [x] revert the PR and gradually add back the `openssl` improvements we intended, by monitoring the ecosystem-tests

**Comment by jkomyno** [comment]
- Vercel uses two different systems (one for building the serverless function, one for executing it) with different `libssl` versions available. The only compatible version between the two is `openssl-1.0.x`.
- We had attempted to prefer the newest versions available of `libssl` (as using a total order helps reducing ambiguity, and newer versions are safer to use), but we reverted that change due to Vercel.
